### PR TITLE
GEOMESA-1764 Docs: improve mirroring of tutorial/nifi READMEs to docs

### PR DESCRIPTION
**Pull request by mdzimmerman**
* Add imported Nifi readme
* Add imported tutorial readmes
* Update scripts with command-line switches to specify paths
* Infer GeoMesa path by default

Signed-off-by: Matthew Zimmerman <matt.zimmerman@ccri.com>

### CODE REVIEW

File: docs/bin/common.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    cd $dir
+    n=`git remote -v show 2>&1 | grep "$url" | wc -l`
+    cd $cwd
+    if [[ "$n" -gt 0 ]] ; then
+        return 0
+    else
+        echo "ERROR: $dir does not have specified remote url"
+        exit 3
+    fi
+}
```
Comment (aheyne) [inline docs/bin/common.sh]: [picky] new line

File: docs/bin/import-nifi-readme.sh [hunk 2]
```diff
@@ -8,15 +8,70 @@
 # have to hand-edit the files after this import!
```
Comment (aheyne) [inline docs/bin/import-nifi-readme.sh]: Are these warning still relevant?

File: docs/bin/common.sh (hunk 1 above)
Comment (aheyne) [inline docs/bin/common.sh]: [Picky] newline

File: docs/bin/import-tutorial-readme.sh [hunk 3]
```diff
@@ -8,9 +8,55 @@
...
+       *) # end of options
+           break
+           ;;
+   esac
+done
+
+#
 if [ "$#" -eq 0 ] ; then
     TUTORIALS="
 geomesa-examples-authorizations
```
Comment (aheyne) [inline docs/bin/import-tutorial-readme.sh]: Probably need to add the geomesa-quickstart-nifi to this list.

File: docs/bin/import-tutorial-readme.sh [hunk 4]
```diff
@@ -27,19 +73,30 @@ else
...
+    text_dst=$GEOMESA/docs/tutorials/${tutorial}.rst
     echo "Converting $text_src => $text_dst"
     pandoc -i $text_src -o $text_dst
 
     # copy the assets too, if they exist
-    asset_src=$TUTORIAL_PATH/assets/${tutorial}
-    asset_dst=$GEOMESA_PATH/docs/tutorials/_static/${tutorial}
+    asset_src=$GEOMESA_TUTORIALS/assets/${tutorial}
+    asset_dst=$GEOMESA/docs/tutorials/_static/${tutorial}
     if [ -d $asset_src ] ; then
```
Comment (aheyne) [inline docs/bin/import-tutorial-readme.sh]: [[Picky]] double brackets

File: docs/tutorials/geomesa-examples-authorizations.rst [hunk 5]
```diff
@@ -454,7 +457,7 @@ Configure GeoServer for PKI Login
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Follow the instructions located
-`here <http://docs.geoserver.org/stable/en/user/security/tutorials/cert/index.html>`__
+`here <http://docs.geoserver.org/2.9.1/user/security/tutorials/cert/index.html>`__
```
Comment (aheyne) [inline docs/tutorials/geomesa-examples-authorizations.rst]: Is there a reason we're binding this to a version?

File: docs/tutorials/geomesa-examples-authorizations.rst [hunk 6]
```diff
@@ -498,7 +504,9 @@ We want to configure LDAP with a user to match the Spring Security PKIs
 we are testing with. The end result we want is to create the following
 user:
 
-``DN: cn=rod,ou=Spring Security,o=Spring Framework``
+::
```
Comment (aheyne) [inline docs/tutorials/geomesa-examples-authorizations.rst]: Seems odd...

File: docs/bin/import-nifi-readme.sh (hunk 2 above)
Comment (mdzimmerman) [inline docs/bin/import-nifi-readme.sh]: Yes. I've tried to minimize the number of conversions that need to be hand-edited, but some are difficult to get rid of. For example, some links in the tutorial and nifi README.md files are manual links to geomesa.org or relative links to other READMEs that should be converted to ``:ref:``, ``:doc:``, or ``:download:`` directives.

File: docs/bin/import-tutorial-readme.sh (hunk 3 above)
Comment (mdzimmerman) [inline docs/bin/import-tutorial-readme.sh]: Oh, nice!

It doesn't look like the geomesa-quickstart-nifi PR has been merged into the master of geomesa-tutorials yet; I'll review that after this.

File: docs/tutorials/geomesa-examples-authorizations.rst (hunk 5 above)
Comment (mdzimmerman) [inline docs/tutorials/geomesa-examples-authorizations.rst]: Yeah, nominally 2.9.1 is the version of GeoServer that GM 1.3.x (and thus the tutorials) supports, but GeoServer stable is already on 2.11.

File: docs/tutorials/geomesa-examples-authorizations.rst (hunk 6 above)
Comment (mdzimmerman) [inline docs/tutorials/geomesa-examples-authorizations.rst]: That's the shortened form of a ``.. code:`` directive, but I can change it.

**Comment by jnh5y** [comment]
@mdzimmerman is this PR good to go?  @elahrvivaz @aheyne anything else to sort out?

**Comment by aheyne** [comment]
Looks like @mdzimmerman may have a commit to push but otherwise it's probably fine.

File: docs/bin/import-tutorial-readme.sh (hunk 4 above)
Comment (mdzimmerman) [inline docs/bin/import-tutorial-readme.sh]: Fixed.

File: docs/bin/common.sh (hunk 1 above)
Comment (mdzimmerman) [inline docs/bin/common.sh]: Fixed.

File: docs/bin/import-tutorial-readme.sh (hunk 3 above)
Comment (mdzimmerman) [inline docs/bin/import-tutorial-readme.sh]: I've merged both the Nifi README and the Nifi Quickstart.

File: docs/tutorials/geomesa-examples-authorizations.rst (hunk 6 above)
Comment (mdzimmerman) [inline docs/tutorials/geomesa-examples-authorizations.rst]: Fixed.

**Comment by mdzimmerman** [comment]
Alright, I've responded to @aheyne 's comments, and imported both the nifi README and the nifi quickstart tutorials. Also fixed a typo in the Cassandra chapter.

**Pull request merged**